packages/synapse-react-client/src/utils/functions/TableColumnSchemaUtils.ts

```diff
@@ -47,23 +74,34 @@ export async function createTableUpdateTransactionRequest(
     .list
 
   // now that all columns have been created, figure out the column changes (create, update, and no-op)
-  // keep track of column ids to figure out what columns were deleted.
   const changes: ColumnChange[] = []
-  const columnIds: Set<string | null> = new Set()
+  const columnIdsThatWereChangedOrAdded: Set<string> = new Set()
   for (let i = 0; i < proposedSchema.length; i++) {
-    const oldColumnId: string | null = proposedSchema[i].id ?? null
-    const newColumnId: string | null = createdColumnModels[i].id ?? null
-    columnIds.add(oldColumnId)
-    columnIds.add(newColumnId)
-    if (oldColumnId !== newColumnId) {
-      changes.push({ oldColumnId, newColumnId })
+    const columnIdInProposedSchema: string | null = proposedSchema[i].id ?? null
+    const newColumnId: string = createdColumnModels[i].id
+    if (columnIdInProposedSchema)
+      columnIdsThatWereChangedOrAdded.add(columnIdInProposedSchema)
```

Comment:
```suggestion
      // We add the columnIdInProposedSchema because the column was modified. It should not be deleted.
      columnIdsThatWereChangedOrAdded.add(columnIdInProposedSchema)
```